src/viewer.js

```diff
@@ -143,6 +143,14 @@ export class Viewer {
     /** @const @private {!Object<string, string>} */
     this.params_ = {};
 
+    /** @private {?function(*)} */
+    this.onVisibilityResolve_ = null;
+
+    /** @private @const {!Promise} */
+    this.onVisibilityPromise_ = new Promise((resolve) => {
+      this.onVisibilityResolve_ = resolve;
```

Comment:
when should that check happen? can we make that `isVisible` check inside `buildCallback`?